docs/modules/nifi/pages/usage_guide/monitoring.adoc

```diff
@@ -34,94 +35,83 @@ spec:
-tls_config:
-  insecure_skip_verify: true
-static_configs:
-  - targets:
-    - '<pod>.<statefulset>.svc.cluster.local:8443' <3>
-metrics_path: '/nifi-api/flow/metrics/prometheus'
-scheme: https
-----
-<1> Use the `authorization` property instead if the `basic_auth`.
-<2> Add the previously obtained token here.
-<3> Static targets only scrapes one pod.
+IMPORTANT: If the {prometheus-operator}[Prometheus Operator,window=_blank] is being used to deploy Prometheus, there is currently a known bug, which prevents adding an additional Volume containing annotations on the volumeClaimTemplate. The bug is tracked in the https://github.com/prometheus-operator/prometheus-operator/issues/7709[Prometheus Operator Issues,window=_blank]. The annotations are necessary to configure the behavior of the Secret Operator. As a current workaround, until the issue is resolved, one could deploy an additional Pod only responsible for creating a TLS certificate as a Secret, which then can be used by the ServiceMonitor. This workaround is illustrated in the https://github.com/stackabletech/demos/blob/main/stacks/monitoring[`monitoring` Stack,window=_blank].
 
-or use it in a NiFi secret which should look like
 [source,yaml]
 ----
 ---
-apiVersion: v1
-kind: Secret
-metadata:
-  name: nifi-authorization-secret
-type: Opaque
-stringData:
-  nifi_token: "<Bearer_token>"
+prometheus: <1>
+  prometheusSpec:
+    volumes:
+      - name: tls
+        ephemeral:
+          volumeClaimTemplate:
+            metadata:
+              annotations:
+                secrets.stackable.tech/class: tls # <2>
+                secrets.stackable.tech/scope: pod,service=prometheus-kube-prometheus-prometheus # <3>
+            spec:
+              storageClassName: secrets.stackable.tech
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: "1"
+    volumeMounts:
+      - name: tls
+        mountPath: /stackable/tls # <4>
 ----
+<1> This given configuration is set in the {prometheus-operator}docs/api-reference/api/#monitoring.coreos.com/v1.Prometheus[Prometheus resource,window=_blank] for the {prometheus-operator}[Prometheus Operator,window=_blank]
+<2> We use the `tls` SecretClass here, shipped with the Secret Operator, storing its CA in a Kubernetes Secret. Any other SecretClass can be used as well
+<3> We add the `service=prometheus-kube-prometheus-prometheus` scope for the Prometheus Service to be included in the `subjectAlternateName` of the TLS certificate. This particular Service name, used here, refers to the Prometheus Service deployed by the {prometheus-operator}[Prometheus Operator,window=_blank]
+<4> The path where the mTLS certificates are mounted to inside the Prometheus Pod
 
 If you want to use a `ServiceMonitor` you'd need to configure it as follows:
 
-// TODO: The ServiceMonitor should be switched to the -metrics service
-
 [source,yaml]
 ----
 ---
 apiVersion: monitoring.coreos.com/v1
```

Comment:
I would suggest to copy most parts (exluding tls) of the ServiceMonitor from the working monitoring stack.
E.g. the `replacement` and `regex` are outdated and need to be updated